lib/vector-config/src/schema/visitors/unevaluated.rs

```diff
@@ -342,13 +361,35 @@ fn is_markable_schema(definitions: &Map<String, Schema>, schema: &SchemaObject)
                     .as_ref()
                     .and_then(|reference| {
                         let reference = get_cleaned_schema_reference(reference);
-                        definitions.get(reference)
+                        definitions.get(reference).map(|d| (reference, d))
+                    })
+                    .and_then(|(definition_name, schema_definition)| {
+                        schema_definition.as_object().map(|s| (definition_name, s))
+                    })
+                    .map_or(false, |(name, schema)| {
+                        debug!(
+                            "Following schema reference '{}' for subschema markability.",
+                            name
+                        );
+                        is_markable_schema(definitions, schema)
                     })
-                    .and_then(Schema::as_object)
-                    .map_or(false, is_object_schema)
             })
             .any(identity);
 
+        debug!(
+            "Schema {} object subschema(s) and {} referenced subschemas.",
+            if has_object_subschema {
+                "has"
+            } else {
+                "does not have"
+            },
```

Comment:
`rustfmt` splitting this across so many lines makes it tempting to use `.then.unwrap_or` here:
```suggestion
            has_object_subschema.then("has").unwrap_or("does not have"),
```

Reply:
That is indeed cleaner. 👍🏻

Reply:
Clippy is so nitpicky, jesus:

```
error: use of `.then_some(..).unwrap_or(..)` can be written more clearly with `if .. else ..`
   --> lib/vector-config/src/schema/visitors/unevaluated.rs:382:13
    |
382 | /             has_referenced_object_subschema
383 | |                 .then_some("has")
384 | |                 .unwrap_or("does not have"),
    | |___________________________________________^ help: try: `if has_referenced_object_subschema { "has" } else { "does not have" }`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#obfuscated_if_else
```

I tried. 😂

Reply:
😢